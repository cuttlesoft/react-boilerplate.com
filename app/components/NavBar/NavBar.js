/* eslint-disable react/prop-types */
import React, { useContext } from 'react'
import { User } from 'grommet-icons'
import { observer } from 'mobx-react'
import { withRouter } from 'react-router-dom'

// Components
import { Avatar } from '../Avatar'
import { Box } from '../Box'
import { Header } from '../Header'
import { Heading } from '../Heading'
import { Link } from '../Link'
import { Menu } from '../Menu'

// Helpers
import { doc } from './NavBar.doc'
import { createWithDoc } from '../../utils/helpers'

// Stores
import { RootStoreContext } from '../../stores/RootStore'

/**
 *
 * NavBar
 *
 */
const NavBar = observer(
  withRouter(({ history, ...rest }) => {
    const { clearStore, user } = useContext(RootStoreContext)

    return (
      <Header background="light-2" pad="small" {...rest} style={{ width: '100vw' }}>
        <Box>
          <Link to={user.isAuthenticated ? '/dashboard' : '/'}>
            <Heading level="2">React Boilerplate</Heading>
          </Link>
        </Box>

        {user.isAuthenticated && (
          <Menu
            dropProps={{ align: { top: 'bottom', left: 'left' } }}
            label={
              <Avatar background="brand">
                <User color="white" />
              </Avatar>
            }
            items={[
              {
                label: 'Settings',
                onClick: () => history.push('/profile'),
              },
              {
                label: 'Logout',
                onClick: () => {
                  clearStore()
                },
              },
            ]}
          />
        )}
      </Header>
    )
  }),
)

export default createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: NavBar,
})
