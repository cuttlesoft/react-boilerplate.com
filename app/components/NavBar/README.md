<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## withRouter()
A NavBar

## Usage

```javascript
import { NavBar } from 'components/NavBar';
<NavBar />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  