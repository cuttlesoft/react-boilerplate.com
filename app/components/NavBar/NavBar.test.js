import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import NavBar from './NavBar'

const renderComponent = (props = {}) =>
  render(
    <MemoryRouter initialEntries={['/']}>
      <NavBar {...props} />
    </MemoryRouter>,
  )

/**
 *
 * Tests for NavBar
 *
 */
describe('NavBar', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
