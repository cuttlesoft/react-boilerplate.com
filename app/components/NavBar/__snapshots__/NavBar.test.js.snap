// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders and matches snapshot 1`] = `
<div>
  <header
    class="StyledBox-sc-13pk1d4-0 ixgMvS"
    style="width: 100vw; transition: none;"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 bFwQzJ"
      style="transition: all .3s ease-in-out;"
    >
      <a
        class="Link__StyledLink-sc-1vpl4pu-0 gEErvb"
        href="/"
        theme="[object Object]"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 fjsCgo"
          style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
        >
          React Boilerplate
        </h2>
      </a>
    </div>
  </header>
</div>
`;
